clinica/iotools/converters/adni_to_bids/adni_utils.py

```diff
@@ -1582,3 +1580,17 @@ def remove_tmp_dmc_folder(bids_dir, image_id):
     tmp_dcm_folder_path = join(bids_dir, f"tmp_dcm_folder_{str(image_id).strip(' ')}")
     if exists(tmp_dcm_folder_path):
         rmtree(tmp_dcm_folder_path)
+
+
+def load_clinical_csv(clinical_dir: str, filename: str) -> pd.DataFrame:
+    import re
+    from pathlib import Path
+
+    pattern = filename + "(_\d{1,2}[A-Za-z]{3}\d{4})?.csv"
+    for z in Path(clinical_dir).rglob("*.csv"):
+        if re.search(pattern, (z.name)):
+            adni_merge_path = z
+    try:
+        return pd.read_csv(adni_merge_path, sep=",", low_memory=False)
+    except:
+        raise ValueError(f"{filename}.csv was not found. Please check your data.")
```

Comment:
I would be a bit more specific here. If you get to this line, the file exists but the reading as a CSV file failed:

```python
raise ValueError(
    f"File {adni_merge_path} was found but could not "
    "be loaded as a DataFrame. Please check your data."
)
```

Reply:
That is not actually what happens. The file is actually not found, and it errors because `adni_merge_path` is therefore not defined.

Reply:
I meant, if you implement the suggestion above, the file has to exist otherwise an error would have been raised before. So you would get to this line only if the CSV file wasn't formatted as expected. Right ?

Reply:
My bad, I have been reading from bottom to top, I'll come back afterwards

Reply:
Done as well, works better reading from the top